### DISCUSSION TITLE
External link icon

### DISCUSSION THREAD

**Discussion by sarahill** [Shopify/polaris-react#6558] [General] [answered]
The external link icon has been a recurring topic of conversation. There is confusion around when and when not use it and questions around the general usefulness of it.

**Issues:**

- The icon is used inconsistently
- The icon adds more visual load especially when used inside a sentence or accompanied by other content
- External links don't always need to be called out
- Current guidance for using the icon has been to use only when the link takes you outside of the Shopify ecosystem but this logic starts to break down pretty quick (example: help center). Adding rules and logic around when and when not use the icon is a slippery slope and makes it difficult to have consistency in how it's used.

**Opportunities:**

We can simplify and add clarity to external links by relying less on the external link icon and focusing more on what the text itself conveys and at what point in their workflow a merchant is encountering them.

- Remove the external link icon. We don't need the icon for an external link to be accessible. The hint text will still be available when using the `external` prop for a link so that it works with screen readers.
- If it is a case where it's important to let the merchant know they're navigating away from the Shopify admin, you can do this through the link text itself

**Related issues:**

- https://github.com/Shopify/polaris/issues/4547
- https://github.com/Shopify/polaris/issues/4756
- https://github.com/Shopify/polaris/issues/6413

I'd like to use this discussion to collect others thoughts and opinions on this topic.

**Comment by alex-page** [comment]
Gov UK said it better then I will in 2016.

> **You don’t always need to tell people about external links**
> There are some cases where you might want to warn users about external links in advance (for example if they’re in the middle of a task), but there’s no need to tell them every time.
>
> **The icon is obscure and ambiguous**
> The icon is not widely recognised and is often confused with the ‘opens in a new window’ icon.
> 
> Some websites actually use it to mean this, so it’s not an effective way of telling people that a link goes to another website.
> 
> **Words are more effective**
> If your users do need to know in advance that a link goes to another website, it’s more effective to do this with words.
>
> You can do this in the link text itself, or if you have a collection of links you might want to do this in the immediately preceding text to avoid duplication.

https://designnotes.blog.gov.uk/2016/11/28/removing-the-external-link-icon-from-gov-uk/

Links should be clear where they end up. An icon doesn't really help with that. It either goes where a user expects or not.

**Comment by m-meurer** [comment]
Would all links open in the same tab by default? This is personally frustrating for me here in Github. I don't necessarily need a visual indication that a link would open in a new tab, but I definitely don't want external links opening in the same tab where I might be doing work in the Admin.

> The external prop adds an icon and a notification that **the link opens a new tab** (emphasis added). Use the external prop to make the link open in a new tab (or window, depending on the merchant’s browser settings). Open a page in a new tab only when opening a page in the same tab might disrupt the merchant’s workflow.

**Reply by sarahill** [reply]
Great question! 

A link by default opens in the same tab but if I add the `external` property to it, it will open in a new tab/window. That's the behavior today and that won't change.

The only change would be the removal of the icon. So there would still be the `external` prop with a notification for screen readers that the link will open in a new tab.

**Reply by alex-page** [reply]
Do we need the screen reader notification if the user knows the link and the URL? I was confused why we added this.

**Reply by sarahill** [reply]
If you're visually impaired I think it's useful to have the hint so the screen reader can call it out. To your point though I think if it's a case where we think we need to be explicit about it the link text should do that job.

**Comment by heyjoethomas** [comment]
This makes a lot of sense to me. Seems like a win

**Comment by kaivab** [comment]
Hi! I'm a content designer on the Retail team - Emily shared this in the content design channel for feedback/discussion. In Polaris for Retail, we currently use the external link icon on any buttons or links that take you outside of the POS app (examples: opening the store's website, or navigating to Shopify admin). Do you think the app use case is different enough that it warrants the external link? Or should we also considering dropping ours?

**Reply by sarahill** [reply]
>  we currently use the external link icon on any buttons or links that take you outside of the POS app (examples: opening the store's website, or navigating to Shopify admin)

Hey! Thanks for this question, it's a good one.

In Polaris we've made such a close tie to the `external icon` and `external links` that the difference between them gets fuzzy. What you're describing above feels like it follows our recommendation for how to use `external links`, which in general is: use an external link when navigating to a page that's outside the Shopify admin. The only change here is dropping the icon which we feel doesn't add much benefit. All that to say, I think you could drop the icon and keep the way you are using external links the same. Let me know if that makes sense or not.

This post from Gov.uk does a good job of talking about why they decided to remove the icon. https://designnotes.blog.gov.uk/2016/11/28/removing-the-external-link-icon-from-gov-uk/

**Reply by steve-howe** [reply]
Hi there! I also work on the POS app along with Kaiva. 

I completely understand the rationale for removing external link icons, particularly for websites. If I'm on a web page, I expect that any link I click on might take me _anywhere_ else on the web. It's not a surprise if I leave a particular site. 

I feel like apps are slightly different - and that POS is slightly different, too. The POS app is often the only app (or one of few) running on a particular device. A physical retail store might have a laptop for accessing Shopify admin/the rest of the web, and a tablet with the Shopify POS app running constantly for staff to build carts, process payments, etc. In this context, knowing that a link takes you out of the app can be quite useful. Staff may decide that leaving the app will slow down their workflow and decide against it. 

This is a complete assumption on my part, and to be honest I don't think removing the icons will cause a huge amount of disruption. I'm 100% in favour of simplification, I just want to make sure that context beyond admin/web-based products has been considered.

**Reply by sarahill** [reply]
> I feel like apps are slightly different - and that POS is slightly different, too. The POS app is often the only app (or one of few) running on a particular device. A physical retail store might have a laptop for accessing Shopify admin/the rest of the web, and a tablet with the Shopify POS app running constantly for staff to build carts, process payments, etc. In this context, knowing that a link takes you out of the app can be quite useful. Staff may decide that leaving the app will slow down their workflow and decide against it.

That's fair. The rationale of "it might slow me down" is interesting one and makes sense. Couple questions that come to mind:
- Do merchants and staff understand what the icon means?
- Are there other ways you can signal the link will open in a new tab instead of using the icon? 

@jjgali curious if you have an opinion here or ran into anything on this when you were looking into external links?

**Reply by steve-howe** [reply]
_"Do merchants and staff understand what the icon means?"_

To be honest, I don't know! It's not something we've tested. 

_"Are there other ways you can signal the link will open in a new tab instead of using the icon?"_

Yes, for sure. For example, we'd normally call out "Shopify Help Center" rather than just say "Get support". We can definitely think about how to do that for other external destinations. 

To be clear, I can see the benefit of removing the icon. I've definitely come across the problem of not knowing whether to use the icon or not (for example, if you go to Shopify admin from POS -> is that "external"?) So this definitely removes that ambiguity.

**Reply by sarahill** [reply]
For sure! The questions were more out of curiosity than anything. I trust you all to make the best decision for your use case. I really appreciate you all sharing your perspectives and thoughts

**Reply by jjgali** [reply]
> curious if you have an opinion here or ran into anything on this when you were looking into external links?

I don't have any additional information apart from what @sara shared, and I agree that POS on an app is divergent enough from the admin experience to warrant its own consideration. 

As Sara said, I'd find out if users actually understand what the icon implies. Our inconsistent implementation in the admin shows that even we (Shopify UXers collectively) have a hard time defining the icon's purpose. 

Also, +1 to being as explicit as possible about what/where the resource is you're linking to. If we could avoid the ambiguity that the icon introduces, that'd be ideal.

Would love to know what you end up deciding on!

**Reply by alex-page** [reply]
Point of Sale has it's own users and the experience is unique. I don't see all Polaris guidance needing to scale to Point of Sale. I think this is a case where you can make the best decision for your users. They might benefit from aligning to Shopify Admin or it might be different.

**Comment by sarahill** [comment]
Hey folks! Just wanted to give an update on progress for this. 

We now have a [PR](https://github.com/Shopify/polaris/pull/6518) up to remove the external link icon along with updating the usage guidance.

One thing we would like feedback on is the decision to remove visually hidden hint text along with the icon. After digging into this and having discussion with folks, we are leaning toward removing the visually hidden text that was for screen readers only. Some context behind this decision:

- We believe it may actually degrade the experience instead of enhancing it. The rationale for removing the icon is to remove ambiguity and improve the experience through clear link text. The screen reader only text feels like the equivalent of the icon here. If we make the text clear, that should translate for sited and non sited users.

That being said, we want to get a gut check on this decision. Please share any thoughts, concerns, insights you have. Thank you!

**Reply by alex-page** [reply]
+1 I agree with this rationale. It would be annoying as a screen reader user to hear "opens in a new window" or "external link" when going through a website.

**Reply by heyjoethomas** [reply]
Do we have any native screen reader users at Shopify we can consult with? Would love to hear if there is a standard for things like this.

**Reply by alex-page** [reply]
Not that I know of. We do have accessibility specialists and services to connect with these folks. The GovUK made this change and it was thoroughly tested with users of assistive technologies.

**Reply by sarahill** [reply]
Our next step is to reach out in the accessibility channel for feedback

**Reply by svinkle** [reply]
Please reach out to [#blind-at-shopify](https://shopify-comm.slack.com/archives/C02C8TN0LE8).

**Reply by emily-broughton** [reply]
@svinkle , done! Thanks for the reco!

**Comment by joellesasseville** [comment]
Hello @sarahill 
I do agree with the rational about links. But what about actions that need to be performed in other systems or tool? 

Working in SFN, we often times need to link out to 3rd party or to open a new tab to get merchants perform a task (book a call in ChiliPiper, Fix error in the bulk editor, which cannot be loaded in the same page). From my (probably totally) biased perspective, the icon sets the expectation that this action will not take place in the same context. 

More context : [https://github.com/Shopify/polaris/issues/6799](https://github.com/Shopify/polaris/issues/6799)

**Reply by alex-page** [reply]
Buttons going to external pages is an antipattern. I don't think we should scale this. Clicking a button should be a high trust moment.

In your situation what is happening? Is the page that the button takes the user to look and work the same as the admin? Does the user know they are going to a new website? Does the experience change dramatically?

**Reply by alex-page** [reply]
Looking at the screenshots I believe these should be links and not buttons.

**Reply by joellesasseville** [reply]
I do agree this isn't a perfect solution. In an ideal world, we'd be able to open the bulk editor in the same page, and we would have our own call scheduling system. This would allow keep the merchant workflow streamlined and consistent with the admin experience, but this isn't something we can't do at the moment. 

The screenshots might not provide enough context, but these buttons are key actions that merchants need to perform. Links wouldn't represent well their level of importance in the flow and would be inconsistent with the other steps.

Usually buttons links to actions that are performed in the admin, but it is not our case. We felt it was important to set expectations that these actions behave differently.

**Reply by sarahill** [reply]
I think that's a fair argument and I agree there are some cases where we can use button styling to reinforce hierarchy. I'm still not 100% convinced that the icon is as useful here but this might be something to post for feedback in the [#accessibility ](https://shopify.slack.com/archives/C03T74ZTD) channel. @joellesasseville are you down to post in that channel? I would be interested to hear the feedback from those folks.

**Reply by sarahill** [reply]
Just an update on this: We shared this discussion in a couple different channels for feedback

**Comment by sarahill** [comment, marked as answer]
The updated guidance on external links has now been merged in https://github.com/Shopify/polaris/pull/6518

The PR:
- removes the external link icon from the Link components `external` prop
- updates the guidance for external links

Thanks for everyone's questions and feedback in this discussion!
